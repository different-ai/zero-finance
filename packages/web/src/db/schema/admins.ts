import { pgTable, text, timestamp } from 'drizzle-orm/pg-core';

/**
 * Admins table - stores authorized admin users
 * Access to admin panel is granted only to users in this table
 */
export const admins = pgTable('admins', {
  privyDid: text('privy_did').primaryKey(),
  createdAt: timestamp('created_at', { withTimezone: true })
    .defaultNow()
    .notNull(),
  addedBy: text('added_by'), // Privy DID of admin who added this admin
  notes: text('notes'), // Optional notes about this admin
});

export type Admin = typeof admins.$inferSelect;
export type NewAdmin = typeof admins.$inferInsert;
