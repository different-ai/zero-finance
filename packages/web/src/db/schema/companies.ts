// Placeholder for companies module - will be extracted in future PR
// For now, re-export from main schema to avoid circular dependencies
export { companies } from '../schema';
