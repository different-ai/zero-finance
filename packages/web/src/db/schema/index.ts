// Export from domain modules
export * from './users';
export * from './workspaces';
export * from './admins';

// Re-export everything else from main schema, excluding what we've already exported above
export {
  // Invoice tables
  userWalletsTable,
  userProfilesTable,
  userRequestsTable,
  invoiceTemplates,
  userInvoicePreferences,
  userProfilesRelations,
  userWalletsRelations,

  // Invoice types
  type InvoiceRole,
  type InvoiceStatus,
  type UserWallet,
  type NewUserWallet,
  type UserProfile,
  type NewUserProfile,
  type UserRequest,
  type NewUserRequest,
  type InvoiceTemplate,
  type NewInvoiceTemplate,
  type UserInvoicePreferences,
  type NewUserInvoicePreferences,

  // Safe tables
  userSafes,
  userSafesRelations,
  type UserSafe,
  type NewUserSafe,

  // Banking tables
  userFundingSources,
  userDestinationBankAccounts,
  offrampTransfers,
  onrampTransfers,
  allocationStrategies,
  userFundingSourcesRelations,
  userDestinationBankAccountsRelations,
  offrampTransfersRelations,
  onrampTransfersRelations,
  allocationStrategiesRelations,
  type UserFundingSource,
  type NewUserFundingSource,
  type UserDestinationBankAccount,
  type NewUserDestinationBankAccount,
  type OfframpTransfer,
  type NewOfframpTransfer,
  type OnrampTransfer,
  type NewOnrampTransfer,
  type AllocationStrategy,
  type NewAllocationStrategy,

  // Earn tables
  earnDeposits,
  earnWithdrawals,
  earnVaultApySnapshots,
  incomingDeposits,
  autoEarnConfigs,
  type EarnDeposit,
  type NewEarnDeposit,
  type EarnWithdrawal,
  type NewEarnWithdrawal,
  type EarnVaultApySnapshot,
  type NewEarnVaultApySnapshot,
  type IncomingDeposit,
  type NewIncomingDeposit,

  // Chat tables
  chats,
  chatMessages,
  chatsRelations,
  chatMessagesRelations,
  type ChatDB,
  type NewChatDB,
  type ChatMessageDB,
  type NewChatMessageDB,

  // Classification
  userClassificationSettings,
  userClassificationSettingsRelations,
  type UserClassificationSetting,
  type NewUserClassificationSetting,

  // Companies
  companies,
  companyMembers,
  sharedCompanyData,
  companyClients,
  companyInviteLinks,
  companiesRelations,
  companyMembersRelations,
  sharedCompanyDataRelations,
  companyInviteLinksRelations,
  type Company,
  type NewCompany,
  type CompanyMember,
  type NewCompanyMember,
  type SharedCompanyData,
  type NewSharedCompanyData,
  type CompanyClient,
  type NewCompanyClient,
  type CompanyInviteLink,
  type NewCompanyInviteLink,

  // Platform
  platformTotals,
  type PlatformTotal,
  type NewPlatformTotal,

  // Features
  userFeatures,
  type UserFeature,
  type NewUserFeature,
} from '../schema';
