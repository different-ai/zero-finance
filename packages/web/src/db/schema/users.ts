import {
  pgTable,
  text,
  timestamp,
  boolean,
  uuid,
  index,
} from 'drizzle-orm/pg-core';
import { relations } from 'drizzle-orm';

export const users = pgTable(
  'users',
  {
    privyDid: text('privy_did').primaryKey(),
    createdAt: timestamp('created_at', { withTimezone: true })
      .defaultNow()
      .notNull(),
    firstName: text('first_name'),
    lastName: text('last_name'),
    companyName: text('company_name'),
    beneficiaryType: text('beneficiary_type', {
      enum: ['individual', 'business'],
    }),
    alignCustomerId: text('align_customer_id').unique(),
    kycProvider: text('kyc_provider', { enum: ['align', 'other'] }),
    kycStatus: text('kyc_status', {
      enum: ['none', 'pending', 'approved', 'rejected'],
    }).default('none'),
    kycFlowLink: text('kyc_flow_link'),
    alignVirtualAccountId: text('align_virtual_account_id'),
    kycMarkedDone: boolean('kyc_marked_done').default(false).notNull(),
    kycSubStatus: text('kyc_sub_status'),
    kycNotificationSent: timestamp('kyc_notification_sent', {
      withTimezone: true,
    }),
    kycNotificationStatus: text('kyc_notification_status', {
      enum: ['pending', 'sent', 'failed'],
    }),
    loopsContactSynced: boolean('loops_contact_synced')
      .default(false)
      .notNull(),
    userRole: text('user_role', {
      enum: ['startup', 'contractor'],
    })
      .default('startup')
      .notNull(),
    contractorInviteCode: text('contractor_invite_code'),
    primaryWorkspaceId: uuid('primary_workspace_id'),
  },
  (table) => ({
    primaryWorkspaceIdx: index('users_primary_workspace_idx').on(
      table.primaryWorkspaceId,
    ),
  }),
);

export type User = typeof users.$inferSelect;
export type NewUser = typeof users.$inferInsert;
