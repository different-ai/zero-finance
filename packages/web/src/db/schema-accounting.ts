import {
  pgTable,
  text,
  timestamp,
  varchar,
  uuid,
  boolean,
  jsonb,
  numeric,
  index,
  integer,
  date,
  uniqueIndex,
} from 'drizzle-orm/pg-core';
import { relations } from 'drizzle-orm';
import { inboxCards } from './schema';

// Accounting system sources
export type AccountingSource = 'xero' | 'quickbooks' | 'manual' | 'bank_feed';

// Transaction types
export type TransactionType =
  | 'invoice' // Bill from supplier
  | 'bill_payment' // Payment to supplier
  | 'receipt' // Money received
  | 'expense' // Direct expense
  | 'journal' // Journal entry
  | 'transfer' // Bank transfer
  | 'credit_note' // Credit note
  | 'prepayment' // Prepayment
  | 'overpayment'; // Overpayment

// Transaction status in accounting system
export type TransactionStatus =
  | 'draft'
  | 'submitted'
  | 'authorised'
  | 'paid'
  | 'voided'
  | 'deleted';

// Reconciliation status
export type ReconciliationStatus =
  | 'unmatched' // No match found
  | 'suggested' // System suggested match
  | 'matched' // Manually matched
  | 'auto_matched' // Automatically matched
  | 'disputed' // Match disputed
  | 'confirmed'; // Match confirmed

// Accounting Transactions table - stores data from Xero/QuickBooks
export const accountingTransactions = pgTable(
  'accounting_transactions',
  {
    id: uuid('id').defaultRandom().primaryKey(),
    userId: varchar('user_id', { length: 255 }).notNull(),

    // Source system info
    source: text('source').$type<AccountingSource>().notNull(),
    externalId: varchar('external_id', { length: 255 }).notNull(), // ID in Xero/QB
    externalUrl: text('external_url'), // Link to transaction in Xero/QB

    // Transaction details
    transactionType: text('transaction_type')
      .$type<TransactionType>()
      .notNull(),
    transactionNumber: varchar('transaction_number', { length: 100 }), // Invoice/Bill number
    reference: varchar('reference', { length: 255 }), // Reference number

    // Dates
    transactionDate: date('transaction_date').notNull(),
    dueDate: date('due_date'),
    paidDate: date('paid_date'),

    // Parties
    contactName: varchar('contact_name', { length: 255 }),
    contactId: varchar('contact_id', { length: 255 }), // ID in accounting system

    // Financial details
    subtotal: numeric('subtotal', { precision: 20, scale: 2 }),
    taxAmount: numeric('tax_amount', { precision: 20, scale: 2 }),
    totalAmount: numeric('total_amount', { precision: 20, scale: 2 }).notNull(),
    amountPaid: numeric('amount_paid', { precision: 20, scale: 2 }).default(
      '0',
    ),
    amountDue: numeric('amount_due', { precision: 20, scale: 2 }),
    currency: varchar('currency', { length: 3 }).notNull().default('USD'),
    exchangeRate: numeric('exchange_rate', { precision: 10, scale: 6 }).default(
      '1',
    ),

    // Accounting details
    status: text('status').$type<TransactionStatus>().notNull(),
    lineItems: jsonb('line_items'), // Array of line items with GL codes
    glCode: varchar('gl_code', { length: 50 }), // Primary GL code
    glAccountName: varchar('gl_account_name', { length: 255 }),
    taxCode: varchar('tax_code', { length: 50 }),
    trackingCategories: jsonb('tracking_categories'), // Xero tracking or QB classes

    // Bank details
    bankAccountCode: varchar('bank_account_code', { length: 50 }),
    bankAccountName: varchar('bank_account_name', { length: 255 }),

    // Additional metadata
    attachments: jsonb('attachments'), // URLs to attachments
    notes: text('notes'),
    metadata: jsonb('metadata'), // Store additional system-specific data

    // Sync info
    lastSyncedAt: timestamp('last_synced_at').notNull().defaultNow(),
    syncError: text('sync_error'),

    // Timestamps
    createdAt: timestamp('created_at').defaultNow().notNull(),
    updatedAt: timestamp('updated_at').defaultNow().notNull(),
  },
  (table) => ({
    userIdIdx: index('accounting_transactions_user_id_idx').on(table.userId),
    externalIdIdx: uniqueIndex('accounting_transactions_external_id_idx').on(
      table.source,
      table.externalId,
    ),
    transactionDateIdx: index('accounting_transactions_date_idx').on(
      table.transactionDate,
    ),
    statusIdx: index('accounting_transactions_status_idx').on(table.status),
    contactIdx: index('accounting_transactions_contact_idx').on(
      table.contactName,
    ),
  }),
);

// Invoice Reconciliation table - matches inbox cards (invoices) with accounting transactions
export const invoiceReconciliations = pgTable(
  'invoice_reconciliations',
  {
    id: uuid('id').defaultRandom().primaryKey(),
    userId: varchar('user_id', { length: 255 }).notNull(),

    // References
    inboxCardId: varchar('inbox_card_id', { length: 255 }).notNull(), // Reference to inbox card
    transactionId: uuid('transaction_id').references(
      () => accountingTransactions.id,
    ),

    // Match details
    matchStatus: text('match_status')
      .$type<ReconciliationStatus>()
      .notNull()
      .default('unmatched'),
    matchConfidence: numeric('match_confidence', { precision: 5, scale: 2 }), // 0-100
    matchMethod: varchar('match_method', { length: 50 }), // 'manual', 'auto_amount', 'auto_vendor', 'auto_invoice_number', etc.

    // Match criteria used
    matchedOn: jsonb('matched_on'), // { amount: true, vendor: true, date: false, invoice_number: true }

    // Discrepancies
    amountDiscrepancy: numeric('amount_discrepancy', {
      precision: 20,
      scale: 2,
    }),
    dateDiscrepancy: integer('date_discrepancy'), // Days difference
    discrepancyNotes: text('discrepancy_notes'),

    // User actions
    matchedBy: varchar('matched_by', { length: 255 }), // User who made the match
    matchedAt: timestamp('matched_at'),
    confirmedBy: varchar('confirmed_by', { length: 255 }),
    confirmedAt: timestamp('confirmed_at'),
    disputedBy: varchar('disputed_by', { length: 255 }),
    disputedAt: timestamp('disputed_at'),
    disputeReason: text('dispute_reason'),

    // GL Coding (can override transaction coding)
    glCode: varchar('gl_code', { length: 50 }),
    glAccountName: varchar('gl_account_name', { length: 255 }),
    costCenter: varchar('cost_center', { length: 100 }),
    department: varchar('department', { length: 100 }),
    project: varchar('project', { length: 100 }),

    // Approval workflow
    approvalStatus: varchar('approval_status', { length: 50 }).default(
      'pending',
    ), // pending, approved, rejected
    approvedBy: varchar('approved_by', { length: 255 }),
    approvedAt: timestamp('approved_at'),
    approvalNotes: text('approval_notes'),

    // Additional metadata
    metadata: jsonb('metadata'),

    // Timestamps
    createdAt: timestamp('created_at').defaultNow().notNull(),
    updatedAt: timestamp('updated_at').defaultNow().notNull(),
  },
  (table) => ({
    userIdIdx: index('invoice_reconciliations_user_id_idx').on(table.userId),
    inboxCardIdx: index('invoice_reconciliations_inbox_card_idx').on(
      table.inboxCardId,
    ),
    transactionIdx: index('invoice_reconciliations_transaction_idx').on(
      table.transactionId,
    ),
    statusIdx: index('invoice_reconciliations_status_idx').on(
      table.matchStatus,
    ),
    uniqueMatch: uniqueIndex('invoice_reconciliations_unique_match').on(
      table.inboxCardId,
      table.transactionId,
    ),
  }),
);

// Relations
export const accountingTransactionsRelations = relations(
  accountingTransactions,
  ({ many }) => ({
    reconciliations: many(invoiceReconciliations),
  }),
);

export const invoiceReconciliationsRelations = relations(
  invoiceReconciliations,
  ({ one }) => ({
    transaction: one(accountingTransactions, {
      fields: [invoiceReconciliations.transactionId],
      references: [accountingTransactions.id],
    }),
  }),
);

// Helper type for line items
export interface LineItem {
  description: string;
  quantity: number;
  unitPrice: number;
  amount: number;
  taxAmount?: number;
  glCode?: string;
  glAccountName?: string;
  taxCode?: string;
  trackingCategories?: Record<string, string>;
}

// Helper type for attachments
export interface Attachment {
  id: string;
  fileName: string;
  url: string;
  mimeType: string;
  size?: number;
  uploadedAt?: Date;
}

// Helper type for tracking categories (Xero) / Classes (QuickBooks)
export interface TrackingCategory {
  name: string;
  option: string;
}
