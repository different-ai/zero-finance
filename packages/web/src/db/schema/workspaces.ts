import {
  pgTable,
  varchar,
  uuid,
  timestamp,
  boolean,
  uniqueIndex,
  index,
} from 'drizzle-orm/pg-core';
import { relations } from 'drizzle-orm';
import { users } from './users';
import { companies } from './companies';

export const workspaces = pgTable('workspaces', {
  id: uuid('id').primaryKey().defaultRandom(),
  name: varchar('name', { length: 255 }).notNull(),
  createdBy: varchar('created_by', { length: 255 })
    .notNull()
    .references(() => users.privyDid, { onDelete: 'cascade' }),
  createdAt: timestamp('created_at', { withTimezone: true })
    .defaultNow()
    .notNull(),
  updatedAt: timestamp('updated_at', { withTimezone: true })
    .defaultNow()
    .notNull()
    .$onUpdate(() => new Date()),
});

export const workspaceMembers = pgTable(
  'workspace_members',
  {
    id: uuid('id').primaryKey().defaultRandom(),
    workspaceId: uuid('workspace_id')
      .notNull()
      .references(() => workspaces.id, { onDelete: 'cascade' }),
    userId: varchar('user_id', { length: 255 })
      .notNull()
      .references(() => users.privyDid, { onDelete: 'cascade' }),
    role: varchar('role', { length: 50 }).notNull().default('member'),
    joinedAt: timestamp('joined_at', { withTimezone: true })
      .defaultNow()
      .notNull(),
    isPrimary: boolean('is_primary').default(false).notNull(),
  },
  (table) => {
    return {
      workspaceUserIdx: uniqueIndex('workspace_members_workspace_user_idx').on(
        table.workspaceId,
        table.userId,
      ),
      workspaceIdIdx: index('workspace_members_workspace_id_idx').on(
        table.workspaceId,
      ),
      userIdIdx: index('workspace_members_user_id_idx').on(table.userId),
    };
  },
);

export const workspaceInvites = pgTable(
  'workspace_invites',
  {
    id: uuid('id').primaryKey().defaultRandom(),
    workspaceId: uuid('workspace_id')
      .notNull()
      .references(() => workspaces.id, { onDelete: 'cascade' }),
    companyId: uuid('company_id').references(() => companies.id),
    token: varchar('token', { length: 255 }).unique().notNull(),
    createdBy: varchar('created_by', { length: 255 })
      .notNull()
      .references(() => users.privyDid),
    email: varchar('email', { length: 255 }),
    role: varchar('role', { length: 50 }).default('member'),
    shareInbox: boolean('share_inbox').default(true),
    shareCompanyData: boolean('share_company_data').default(true),
    addAsSafeOwner: boolean('add_as_safe_owner').default(false),
    expiresAt: timestamp('expires_at', { withTimezone: true }),
    usedAt: timestamp('used_at', { withTimezone: true }),
    usedBy: varchar('used_by', { length: 255 }).references(
      () => users.privyDid,
    ),
    createdAt: timestamp('created_at', { withTimezone: true })
      .defaultNow()
      .notNull(),
  },
  (table) => {
    return {
      tokenIdx: uniqueIndex('workspace_invites_token_idx').on(table.token),
      workspaceIdx: index('workspace_invites_workspace_idx').on(
        table.workspaceId,
      ),
      createdByIdx: index('workspace_invites_created_by_idx').on(
        table.createdBy,
      ),
    };
  },
);

export const workspaceMembersExtended = pgTable('workspace_members_extended', {
  id: uuid('id').primaryKey().defaultRandom(),
  memberId: uuid('member_id')
    .notNull()
    .references(() => workspaceMembers.id, { onDelete: 'cascade' }),
  canViewInbox: boolean('can_view_inbox').default(true),
  canEditExpenses: boolean('can_edit_expenses').default(false),
  canViewCompanyData: boolean('can_view_company_data').default(true),
});

export const workspacesRelations = relations(workspaces, ({ many, one }) => ({
  members: many(workspaceMembers),
  creator: one(users, {
    fields: [workspaces.createdBy],
    references: [users.privyDid],
  }),
}));

export const workspaceMembersRelations = relations(
  workspaceMembers,
  ({ one }) => ({
    workspace: one(workspaces, {
      fields: [workspaceMembers.workspaceId],
      references: [workspaces.id],
    }),
    user: one(users, {
      fields: [workspaceMembers.userId],
      references: [users.privyDid],
    }),
  }),
);

export const workspaceInvitesRelations = relations(
  workspaceInvites,
  ({ one }) => ({
    workspace: one(workspaces, {
      fields: [workspaceInvites.workspaceId],
      references: [workspaces.id],
    }),
    company: one(companies, {
      fields: [workspaceInvites.companyId],
      references: [companies.id],
    }),
    createdByUser: one(users, {
      fields: [workspaceInvites.createdBy],
      references: [users.privyDid],
    }),
    usedByUser: one(users, {
      fields: [workspaceInvites.usedBy],
      references: [users.privyDid],
    }),
  }),
);

export type Workspace = typeof workspaces.$inferSelect;
export type NewWorkspace = typeof workspaces.$inferInsert;
export type WorkspaceMember = typeof workspaceMembers.$inferSelect;
export type NewWorkspaceMember = typeof workspaceMembers.$inferInsert;
export type WorkspaceInvite = typeof workspaceInvites.$inferSelect;
export type NewWorkspaceInvite = typeof workspaceInvites.$inferInsert;
export type WorkspaceMemberExtended =
  typeof workspaceMembersExtended.$inferSelect;
export type NewWorkspaceMemberExtended =
  typeof workspaceMembersExtended.$inferInsert;
