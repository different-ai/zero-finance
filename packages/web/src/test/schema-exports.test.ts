import { describe, it, expect } from 'vitest';
import * as schema from '../db/schema';

describe('Schema exports snapshot', () => {
  it('exports all expected tables', () => {
    const tables = Object.keys(schema).filter(
      (key) =>
        !key.includes('Relations') &&
        !key.startsWith('New') &&
        typeof schema[key as keyof typeof schema] === 'object' &&
        schema[key as keyof typeof schema] !== null,
    );

    expect(tables.sort()).toMatchSnapshot();
  });

  it('exports all expected types and enums', () => {
    const allKeys = Object.keys(schema);
    const tableKeys = Object.keys(schema).filter(
      (key) =>
        !key.includes('Relations') &&
        !key.startsWith('New') &&
        typeof schema[key as keyof typeof schema] === 'object' &&
        schema[key as keyof typeof schema] !== null,
    );
    const relationKeys = Object.keys(schema).filter((key) =>
      key.includes('Relations'),
    );

    const typeKeys = allKeys.filter(
      (key) => !tableKeys.includes(key) && !relationKeys.includes(key),
    );

    expect(typeKeys.sort()).toMatchSnapshot();
  });

  it('exports all relations', () => {
    const relations = Object.keys(schema).filter((key) =>
      key.includes('Relations'),
    );

    expect(relations.sort()).toMatchSnapshot();
  });

  it('exports specific critical tables', () => {
    const criticalTables = [
      'users',
      'userSafes',
      'userFundingSources',
      'userDestinationBankAccounts',
      'offrampTransfers',
      'onrampTransfers',
      'earnDeposits',
      'earnWithdrawals',
      'workspaces',
      'workspaceMembers',
      'companies',
      'chats',
      'chatMessages',
    ];

    criticalTables.forEach((tableName) => {
      expect(schema).toHaveProperty(tableName);
    });
  });
});
