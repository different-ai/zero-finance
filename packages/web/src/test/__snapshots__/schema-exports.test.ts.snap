// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Schema exports snapshot > exports all expected tables 1`] = `
[
  "allocationStrategies",
  "autoEarnConfigs",
  "chatMessages",
  "chats",
  "companies",
  "companyClients",
  "companyInviteLinks",
  "companyMembers",
  "earnDeposits",
  "earnVaultApySnapshots",
  "earnWithdrawals",
  "incomingDeposits",
  "invoiceTemplates",
  "offrampTransfers",
  "onrampTransfers",
  "platformTotals",
  "sharedCompanyData",
  "userClassificationSettings",
  "userDestinationBankAccounts",
  "userFeatures",
  "userFundingSources",
  "userInvoicePreferences",
  "userProfilesTable",
  "userRequestsTable",
  "userSafes",
  "userWalletsTable",
  "users",
  "workspaceInvites",
  "workspaceMembers",
  "workspaceMembersExtended",
  "workspaces",
]
`;

exports[`Schema exports snapshot > exports all expected types and enums 1`] = `[]`;

exports[`Schema exports snapshot > exports all relations 1`] = `
[
  "allocationStrategiesRelations",
  "chatMessagesRelations",
  "chatsRelations",
  "companiesRelations",
  "companyInviteLinksRelations",
  "companyMembersRelations",
  "offrampTransfersRelations",
  "onrampTransfersRelations",
  "sharedCompanyDataRelations",
  "userClassificationSettingsRelations",
  "userDestinationBankAccountsRelations",
  "userFundingSourcesRelations",
  "userProfilesRelations",
  "userSafesRelations",
  "userWalletsRelations",
  "usersRelations",
  "workspaceInvitesRelations",
  "workspaceMembersRelations",
  "workspacesRelations",
]
`;
